plugins/inputs/mysql/mysql.go

```diff
@@ -609,7 +618,13 @@ func (m *Mysql) parseGlobalVariables(key string, value sql.RawBytes) (interface{
 // This code does not work with multi-source replication.
 func (m *Mysql) gatherSlaveStatuses(db *sql.DB, serv string, acc telegraf.Accumulator) error {
 	// run query
-	rows, err := db.Query(slaveStatusQuery)
+	var rows *sql.Rows
+	var err error
+	if m.MariadbDialect {
+		rows, err = db.Query(slaveStatusQueryMariadb)
+	} else {
+		rows, err = db.Query(slaveStatusQuery)
+	}
```

Comment:
Can we please do this branching once e.g. in `InitMysql()` to make this more simple and readable!?

Reply:
Moved the check to InitMysql()